modules/programs/senpai.nix

```
@@ -0,0 +1,47 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.senpai;
+  cfgFmt = pkgs.formats.yaml { };
+in {
+  options.programs.senpai = {
+    enable = mkEnableOption "senpai";
+    config = mkOption {
+      type = types.submodule {
+        freeformType = cfgFmt.type;
+        options = {
+          addr = mkOption {
+            type = types.str;
+            description = ''
+              The address (host[:port]) of the IRC server. senpai uses TLS connections
+              by default unless you specify no-tls option. TLS connections default
+              to port 6697, plain-text use port 6667.
+            '';
+          };
+          nick = mkOption {
+            type = types.str;
+            description = ''
+              Your nickname, sent with a NICK IRC message. It mustn't contain spaces
+              or colons (:).
+            '';
+          };
+        };
+      };
+      description = ''
+        Configuration for senpai. For a list of options, see
+        <citerefentry><refentrytitle>senpai</refentrytitle>
+        <manvolnum>5</manvolnum></citerefentry>.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.senpai ];
```

Can you add a `package` option that defaults to `pkgs.senpai`?

You can take inspiration from other modules.

```suggestion
    home.packages = [ cfg.package ];
```